src/hooks/useEpochs.ts

```diff
@@ -18,11 +19,18 @@ const useEpochs = () => {
       }
 
       const additionalEpochs = await Promise.all(
-        Array.from({ length: HISTORICAL_EPOCHS_TO_FETCH }, (_, index) => startEpoch.epochIndex - index - 1)
-          .map(epochIndex => getEpoch(arIOReadSDK, epochIndex))
-      );      
+        Array.from(
+          { length: HISTORICAL_EPOCHS_TO_FETCH },
+          (_, index) => startEpoch.epochIndex - index - 1,
+        ).map((epochIndex) =>
+          getEpoch(networkPortalDB, arIOReadSDK, epochIndex),
+        ),
+      );
 
-      return [startEpoch, ...additionalEpochs.filter((e: any) => e !== undefined)];
+      return [
+        startEpoch,
+        ...additionalEpochs.filter((e: any) => e !== undefined),
+      ];
```

Comment:
IDK what getEpoch looks like, but if its a network call and not an async db query I recommend leveraging queryClient.fetchQuery internally in this query and create a cache result for each respective epoch, unless thats leveraged in getEpoch already.

Reply:
It's fetches from db if available and otherwise fetches and caches.